madmom/features/tempo.py

```diff
@@ -720,6 +891,29 @@ def interval_histogram(self, activations):
         return interval_histogram_acf(activations, self.min_interval,
                                       self.max_interval)
 
+    def online_interval_histogram(self, activation):
+        """
+        Compute the histogram of the beat intervals using auto-correlation on
+        buffered activations.
+
+        Parameters
+        ----------
+        activation : numpy float
+            Beat activation for a single frame.
+
+        Returns
+        -------
+        histogram_bins : numpy array
+            Bins of the tempo histogram.
+        histogram_delays : numpy array
+            Corresponding delays [frames].
+        """
+        # shift buffer and put new activation at end of buffer
+        buffer = self._buffer(activation)
+        # use offline acf function on buffered activations
+        return interval_histogram_acf(buffer, self.min_interval,
+                                      self.max_interval)
+
```

Comment:
Yeah, I have to test the ACF performance anyway. If it performs similar to the other methods I will try to change it into a "real" online computation.